src/utils.test.ts

```diff
@@ -1,85 +1,155 @@
-  const result2 = isValidWarnOption(invalid2)
-  const result3 = isValidWarnOption(invalid3)
+  const invalid1 = "rew";
+  const invalid2 = "";
+  const invalid3 = "20g";
+  const result1 = isValidWarnOption(invalid1);
+  const result2 = isValidWarnOption(invalid2);
+  const result3 = isValidWarnOption(invalid3);
 
   expect(result1).toBe(false);
   expect(result2).toBe(false);
   expect(result3).toBe(false);
 });
 
+test("isValidWarnWhenOption validates", () => {
+  const option1 = "1M";
+  const option2 = "2m";
+  const option3 = "3p";
+
+  expect(isValidWarnWhenOption(option1)).toBe(option1);
+  expect(isValidWarnWhenOption(option2)).toBe(option2);
+  expect(isValidWarnWhenOption(option3)).toBe(option3);
+});
+
+test("isValidWarnWhenOption invalidates", () => {
+  const option1 = "reg";
+  const option2 = "";
+  const option3 = "3g";
+
+  expect(isValidWarnWhenOption(option1)).toBe(false);
+  expect(isValidWarnWhenOption(option2)).toBe(false);
+  expect(isValidWarnWhenOption(option3)).toBe(false);
+});
+
+test("parseOption", () => {
+  const dateOption = "20d";
+  const versionOption = "2M";
+  expect(parseOption(dateOption)).toStrictEqual([20, "d"]);
+  expect(parseOption(versionOption)).toStrictEqual([2, "M"]);
+});
+
 test("parseWarnOption", () => {
-  const opt = '23h'
-  const [multipler, interval] = parseWarnOption(opt)
+  const opt = "23h";
+  const [multipler, interval] = parseWarnOption(opt);
 
-  expect(multipler).toBe(23)
-  expect(interval).toBe('h')
-})
+  expect(multipler).toBe(23);
+  expect(interval).toBe("h");
+});
 
 test("parseWarnOption with wrong option", () => {
-  const opt = '5x'
-  const [multipler, interval] = parseWarnOption(opt)
+  const opt = "5x";
+  const [multipler, interval] = parseWarnOption(opt);
 
-  expect(multipler).toBe(1)
-  expect(interval).toBe('w')
-})
+  expect(multipler).toBe(1);
+  expect(interval).toBe("w");
+});
+
+test("parseWarnWhenOption", () => {
+  const option1 = "1M";
+  const option2 = "2m";
+  const option3 = "10p";
+
+  expect(parseWarnWhenOption(option1)).toStrictEqual([1, 0]);
+  expect(parseWarnWhenOption(option2)).toStrictEqual([2, 1]);
+  expect(parseWarnWhenOption(option3)).toStrictEqual([10, 2]);
+});
 
 test("getWarningPeriod with true", () => {
-  const option = true
-  const res = getWarningPeriod(option)
+  const option = true;
+  const res = getWarningPeriod(option);
 
-  expect(res).toBe(604800)
-})
+  expect(res).toBe(604800);
+});
 
 test("getWarningPeriod with interval option", () => {
-  const option = '1h'
-  const res = getWarningPeriod(option)
+  const option = "1h";
+  const res = getWarningPeriod(option);
+
+  expect(res).toBe(3600);
+});
+
+test('getWarningWhen', () => {
+  const option1 = true
+  const option2 = '2m'
+
+  expect(getWarningWhen(option1)).toStrictEqual([1, 2])
+  expect(getWarningWhen(option2)).toStrictEqual([2, 1])
+})
+
+test('getSplitPackageVersion', () => {
+  const p = '1.2.3'
+  expect(getSplitPackageVersion(p)).toStrictEqual([1, 2, 3])
```

Comment:
Did you take into account that people cab also use `2.x` notations for example in their `package.json` file?

Reply:
No, not really! It was quite a ride to get this checking right with the full version description! I think we can have that as an extension!

Reply:
Fair point! I'll create an issue from this comment.

Maybe we can checkout the source code of this elixir credo plugin to check how it's done there!
https://hex.pm/packages/credo_todo_or_die